intrahospital_api/services/base/load_utils.py

```diff
@@ -0,0 +1,160 @@
+from functools import wraps
+from django.utils import timezone
+from django.db import transaction
+from django.db.models import Q
+from opal.models import Patient
+from intrahospital_api import logger
+from intrahospital_api.exceptions import BatchLoadError
+from intrahospital_api import models
+
+MAX_ALLOWABLE_BATCH_RUN_TIME = 3600
+
+
+def any_loads_running():
+    """
+    Returns a boolean as to whether any loads are
+    running, for use by things that synch databases
+    """
+    patient_loading = models.InitialPatientLoad.objects.filter(
+        state=models.InitialPatientLoad.RUNNING
+    ).exists()
+
+    batch_loading = models.BatchPatientLoad.objects.filter(
+        state=models.BatchPatientLoad.RUNNING
+    ).exists()
+    result = patient_loading or batch_loading
+    logger.info("Checking loads are running {}".format(result))
+    return result
+
+
+def good_to_go(service_name):
+    """
+    Are we good to run a batch load, returns True if we should.
+    runs a lot of sanity checks.
+    """
+    batch_loads = models.BatchPatientLoad.objects.filter(
+        service_name=service_name
+    )
+
+    if not batch_loads.exists():
+        return True
+
+    current_running = batch_loads.filter(
+        Q(stopped=None) | Q(state=models.BatchPatientLoad.RUNNING)
```

Comment:
I'm not sure what SQL this outputs ? I don't understand bitwise operator overloading on `Q` objects :(